source/server/server_reactor.h

```diff
@@ -0,0 +1,79 @@
+#ifndef NIDEVICE_GRPC_DEVICE_SERVER_REACTOR_H
+#define NIDEVICE_GRPC_DEVICE_SERVER_REACTOR_H
+
+#include <grpcpp/alarm.h>
+#include <grpcpp/grpcpp.h>
+
+#include <deque>
+#include <memory>
+#include <mutex>
+#include <queue>
+
+namespace nidevice_grpc {
+
+// ServerWriteReactor implementation to support async response streaming.
+template <typename TResponse, typename TProducer>
+class ServerWriterReactor : public grpc::ServerWriteReactor<TResponse> {
+  using LockGuard = std::lock_guard<std::recursive_mutex>;
+
+ public:
+  virtual ~ServerWriterReactor()
+  {
+  }
+
+  void OnDone() override
+  {
+    delete this;
+  }
+
+  void OnWriteDone(bool ok) override
```

Comment:
Is it possible to update this argument name or does the name need to keep in sync with the `grpc::ServerWriteReactor`? If so, something a little more telling might be nice like `continue` or verbosely `start_next_write`.

Reply:
I think `write_succeeded` could be good.

I'm not sure how important it is to match. `ok` is used on the grpc "completion queue" API and overloaded to mean a bunch of different things (depending on what operation you're dequeuing on) that are documented in a sort of intricate way throughout the API. In that context, it's important because the meaning of the variable emerges from all that stuff.

This is definitely inherited from all of that. But in the new API the status is scoped to the specific operation so it seems reasonable to assign it a specific meaning. Having the tie-in to the old API was helpful to me for understanding it, but I'm not sure if that will help anyone else. Hopefully no one else will have to mess with completion queues, though I'm sure we'll run into grpc's `ok` params elsewhere.